code/game/objects/effects/landmarks_events.dm

```diff
@@ -0,0 +1,129 @@
+	var/isRepeating = FALSE
+	var/coordinates = ""
+	var/cooldown = 0 //in seconds
+	var/last_trigger = 0
+	var/isLoud = FALSE
+	var/isNarrate = FALSE
+
+/obj/effect/landmark/event_trigger/New()
+	..()
+	coordinates = "(X:[loc.x];Y:[loc.y];Z:[loc.z])"
+	invisibility = 100
+
+/obj/effect/landmark/event_trigger/proc/set_vars(mob/M)
+	name = sanitize(tgui_input_text(M, "Input Name for the trigger", "Naming", "Event Trigger"))
+	creator_ckey = M.ckey
+	if(!event_triggers[creator_ckey])
+		event_triggers[creator_ckey] = list()
+	event_triggers[creator_ckey] |= list(src)
+	isTeamwork = (tgui_alert(M, "Notify rest of team?", "Teamwork", list("No", "Yes")) == "Yes" ? TRUE : FALSE)
+	if(!isTeamwork)
+		isLoud = (tgui_alert(M, "Should it make a bwoink when triggered for YOU?", "bwoink", list("No", "Yes")) == "Yes" ? TRUE : FALSE)
+	isRepeating = (tgui_alert(M, "Make it fire repeatedly?", "Repetition", list("No", "Yes")) == "Yes" ? TRUE : FALSE)
+	if(isRepeating)
+		cooldown = tgui_input_number(M, "Set cooldown in seconds. Minimum 5 seconds!", "Cooldown", 60, min_value = 5)
+		cooldown = cooldown SECONDS
+	else
+		delete_me = TRUE
+	log_admin("[M.ckey] has created a [isNarrate ? "Narrtion" : "Notification"] landmark trigger at [coordinates]")
+
+/obj/effect/landmark/event_trigger/Destroy()
+	if(event_triggers[creator_ckey])
+		event_triggers[creator_ckey] -= src
+	..()
+
+/obj/effect/landmark/event_trigger/Crossed(var/atom/movable/AM)
+	if(isobserver(AM) || !isliving(AM))
+		return FALSE
+	var/mob/living/L = AM
+	if(!L.ckey) return FALSE
+
+	if(world.time < (last_trigger + cooldown))
+		return FALSE
+	if(!isRepeating && last_trigger) //Used to avoid spam if qdel(src) fires too slowly
+		return FALSE
+	last_trigger = world.time
+
+	if(!creator_ckey)	//For some reason, the user didn't have a ckey. Let's clean up
+		qdel(src)
+		return FALSE
+	var/mob/creator_reference = GLOB.directory[creator_ckey]
+	if(!creator_reference || isTeamwork)	//If logged/crashed, we default to teamwork mode
+		message_admins("Player [L.name] ([L.ckey]) has triggered event narrate landmark [name] of type \
+			[isNarrate ? "Narration" : "Notification"]. \n \
+			The landmark was created by [creator_ckey], and it [isRepeating ? \
+			"will be possible to trigger after [cooldown / (1 SECOND)] seconds" : "has self-deleted"]\n \
+			COORDINATES: [coordinates]")
+	else
+		if(isLoud)
+			creator_reference << 'sound/effects/adminhelp.ogg'
+		to_chat(creator_reference, SPAN_WARNING("Player [L.name] ([L.ckey]) has triggered event \
+			narrate landmark [name] of type [isNarrate ? "Narration" : "Notification"]. \n \
+			It [isRepeating ? "will be possible to trigger after [cooldown / (1 SECOND)] seconds" : "has self-deleted"] \n \
+			COORDINATES: [coordinates]"))
+	if(!isNarrate)
+		if(!isRepeating)
+			qdel(src)
+		return FALSE
+	else
+		return L
+
+
+/obj/effect/landmark/event_trigger/auto_narrate
+	var/message
+	var/isVis_or_isAud = 0	//0 for vis, 1 for aud
```

Comment:
honestly, isVis_or_isAud and isPersonal could all be rolled up into single var with 3 values, 0, 1 and 2 and comparing it to those specific numbers.